client/src/components/AuditAdmin/Progress/JurisdictionDiscrepancies.tsx

```
@@ -0,0 +1,142 @@
+import React from 'react'
+import { Classes, Colors, Dialog, H6, HTMLTable } from '@blueprintjs/core'
+import styled from 'styled-components'
+import {
+  IJurisdiction,
+  useDiscrepanciesByJurisdiction,
+} from '../../useJurisdictions'
+import { useBatches } from '../../JurisdictionAdmin/useBatchResults'
+import useContestsJurisdictionAdmin from '../../JurisdictionAdmin/useContestsJurisdictionAdmin'
+import { IContest } from '../../../types'
+
+const ContestDiscrepanciesTable = styled(HTMLTable).attrs({
+  bordered: true,
+  striped: true,
+})`
+  &.${Classes.HTML_TABLE} {
+    border: 1px solid ${Colors.LIGHT_GRAY1};
+    margin-bottom: 40px;
+    table-layout: fixed;
+    background: #ffffff;
+    width: 100%;
+  }
+
+  &.${Classes.HTML_TABLE} thead tr:not(:last-child) {
+    border-bottom: 1px solid ${Colors.LIGHT_GRAY1};
+    white-space: nowrap;
+  }
+
+  &.${Classes.HTML_TABLE} td {
+    vertical-align: middle;
+  }
+`
+
+function getContestName(contests: IContest[], contestId: string) {
+  const contest = contests.find(c => c.id === contestId)
+  return contest ? contest.name : 'Contest not found'
+}
+
+function getChoiceName(
+  contests: IContest[],
+  contestId: string,
+  choiceId: string
+) {
+  const contest = contests.find(c => c.id === contestId)
+  if (!contest) return 'Choice not found'
+  const choice = contest.choices.find(c => c.id === choiceId)
+  return choice ? choice.name : 'Choice not found'
```

Any suggestion on how I should handle these cases **that we should not see** where the contest choice isn't found?

---

Good points. I'm thinking we could say "Unknown" and then include the relevant ID, so there is at least something to reference. I lean towards thinking it's better than having the page crash, as it doesn't seem dangerous or nonsensical, just unclear. But happy to be overridden on that, still developing the intuition

---

I think these are impossible cases and would recommend using assertions to communicate that. In order for a contest or choice to not be found, there'd need to be a bug in the backend or a caching error. We should fail fast in either case so we can fix it.